### PR TITLE
Improve logging and context handling in basic inttest

### PR DESCRIPTION
**Pull request by twz123**
## Description

This has been done to ease debugging integration test failures when upgrading to Kubernetes 1.27 (#2879):

Refactor CSR checks and add some log statements to better track the progress of the inttest. Also retrieve the context once from the suite and pass it on as usual. This helps to avoid using the suite's context implicitly.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [ ] Auto test added

## Checklist:

- [ ] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [ ] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: inttest/basic/basic_test.go [hunk 1]
```diff
@@ -212,14 +231,14 @@ func (s *BasicSuite) verifyContainerdDefaultConfig() {
 	}).URI(), parsedConfig.Plugins.CRI.SandboxImage)
 }
 
-func (s *BasicSuite) verifyCoreDNSAntiAffinity(kc *kubernetes.Clientset) {
+func (s *BasicSuite) checkCoreDNSAntiAffinity(ctx context.Context, kc *kubernetes.Clientset) error {
```
Comment (mikhail-sakhnov) [inline inttest/basic/basic_test.go:234]: let's either remove the verb "check" from the function name or get assertions s.NoError back? 

having in the name "verify" or "check" imo implies that something would be asserted inside

**Review by mikhail-sakhnov — CHANGES_REQUESTED** [review]
small changes are suggested about the helper naming

File: inttest/basic/basic_test.go (hunk 1 above)
Comment (twz123) [reply inttest/basic/basic_test.go]: I get your point, although I don't really have a super good idea for a better fitting verb. Maybe "probe"?

**Review by mikhail-sakhnov — DISMISSED** [review]
@twz123 it has conflicts now to resolve

**Review by mikhail-sakhnov — APPROVED** [review]

**Pull request merged**